functions/src/modules/SQR.functions.ts

```diff
@@ -265,57 +283,94 @@ export const importSpreadSheetData = functions.https.onRequest( async (req, res)
         scopes: ["https://www.googleapis.com/auth/spreadsheets"]
     });
     
-    const spreadsheetUrl = 'https://docs.google.com/spreadsheets/d/1e0rYguPua_0Rtd--uNLKkGotNw-CPM44TlBS5hthdfQ/edit#gid=477553865';
-    // extracting the spreadsheet ID from the url
-    const spreadsheetId = new RegExp("/spreadsheets/d/([a-zA-Z0-9-_]+)").exec(spreadsheetUrl)[1];
+    
+    const spreadsheetId = functions.config().sqr.spreadsheetId;
     const spreadsheet = new GoogleSpreadsheet(spreadsheetId);
 
     const token = await auth.getAccessToken();
     spreadsheet.setAuthToken(token);
 
-    spreadsheet.getInfo((err, data) => {
-        if (!err) {
-            data.worksheets.forEach(worksheet => {
-                if(worksheet.title === "Submissions")
-                    worksheet.getRows({ }, (err, rows) => {
-                        rows.forEach(row => {
-                            let documentID = row['submissionserial'];
-                            db.ref(`/sqr/submissions/s${documentID}`)
-                                .set({
-                                    devotee: {
-                                        name: row['devotee'],
-                                        emailAddress: row['email'],
-                                    },
-                                    fileName: row['audiofilename'],
-                                    changed: row['changed'],
-                                    completed: row['completed'],
-                                    created: row['created'],
-                                    comments: row['comments'],
-                                    soundissues: row['soundissues'],
-                                    soundqualityrating: row['soundqualityrating'],
-                                    unwantedparts: row['unwantedparts'],
-                                });
+    const getInfo = promisify(spreadsheet.getInfo);
+    let data = await getInfo();
+
+
+    data.worksheets.forEach(async worksheet => {
+        const getRows = promisify(worksheet.getRows);
+
+        if(worksheet.title === "Submissions") {
+            let rows = await getRows({});
+            rows.forEach(row => {
+                let documentID = row['submissionserial'];
+                let spreadsheetRow = {
```

Comment:
Here you are constructing a submission to be added to the database, not a spreadsheet row. Please rename.